src/internal/middleware/logging/interceptor.go

```diff
@@ -361,6 +361,7 @@ func isNilInterface(x interface{}) bool {
 }
 
 func (li *LoggingInterceptor) UnaryServerInterceptor(ctx context.Context, req interface{}, info *grpc.UnaryServerInfo, handler grpc.UnaryHandler) (resp interface{}, retErr error) {
+	ctx = withMethodName(ctx, info.FullMethod)
```

Comment:
BTW, this is kind of built into grpc, but not in an easy to use way.

The full method name is written into the context here: https://github.com/grpc/grpc-go/blob/master/internal/transport/http2_server.go#L574

It is up to you to create a server that is initialized with a StatsHandler that actually puts this into the context.  Here are some implementations of that: https://pkg.go.dev/search?q=TagRPC&m=

TL;DR is that we can probably do all the logging and adding-the-name-to-the-context without an interceptor or custom stream type.  Maybe we should open up a ticket to improve this implementation in general.